scipy/signal/_signaltools.py

```diff
@@ -2452,6 +2456,269 @@ def hilbert2(x, N=None):
     return x
 
 
+def envelope(z: np.ndarray, bp_in: tuple[int | None, int | None] = (1, None), *,
+             n_out: int | None = None, squared: bool = False,
+             residual: Literal['lowpass', 'all', None] = 'lowpass',
+             axis: int = -1) -> tuple[np.ndarray, np.ndarray] | np.ndarray:
+    r"""Compute the envelope of a real- or complex-valued signal.
+
+    Any complex-valued signal `z(t)` can be described by a real-valued instantaneous
+    amplitude `a(t)` and a real-valued instantaneous phase `phi(t)`, i.e.,
+    `z(t) = a(t) * exp(1j*phi(t))`. The envelope is defined as the absolute value of
+    the amplitude `|a(t)| = |z(t)|`, which is at the same time the absolute value of
+    the signal. Hence, `|a(t)|` "envelopes" the class of all signals with amplitude
+    `a(t)` and arbitrary phase `phi(t)`.
+    For real-valued signals, `x(t) = a(t) * cos(phi(t))` is the analogous formulation.
+    `|a(t)|` can be determined by converting `x(t)` into an analytic signal `z_a(t)`
+    by means of a Hilbert transform, i.e.,
+    `z_a(t) = a(t) * cos(phi(t)) + 1j * a(t) * sin(phi(t))`, which produces a
+    complex-valued signal with the same envelope `|a(t)|`.
+
+    Parameters
+    ----------
+    z :
+        Real- or complex-valued input signal, which is assumed to be made up of `n`
+        samples and having sampling interval `T`. `z` may also be a multidimensional
+        signal with the time axis being defined by `axis`.
+    bp_in :
+        2-tuple defining the frequency band ``bp_in[0]:bp_in[1]`` of the input filter.
+        The corner frequencies are specified as multiples of `1/(n*T)` and the allowed
+        frequency range is specified by ``-n//2 <= bp_in[0] < bp_in[1] <= (n+1)//2``.
+        ``None`` entries are replaced with `-n//2` or `(n+1)//2` respectively. The
+        default of ``(1, None)`` removes the mean value as well as the negative
+        frequency components.
+    n_out :
+        If not ``None`` the output will be resampled to `n_out` samples. The default
+        of ``None`` sets the output to the same length as the input `z`.
+    squared :
+        If set, the square of the envelope is returned. This is useful, if filters need
+        to be applied to the result, since the squared envelope possesses fewer
+        harmonics.
```

Comment:
Could you please add an example on this? Especially about its usefulness.

Reply:
Let me put some more thought into this.

Reply:
I was not able to come up with meaningful example which could be described in a single paragraph. I changed the description, which mentions now the effect of the square root in the abs function.